platform/eventing/event.go

```diff
@@ -58,60 +40,28 @@ type Event struct {
 
 // CreatePublisherEvent creates a new events that we'll store inside the publisher_events table.
 // subject: the resource bound to the event (e.g current user id, etc...)
-// eventtype: is the name of the event (e.g `application.created`)
+// type: is the name of the event (e.g `application.created`)
+// source: name of the application that created the event
 // dataschema: is the JSON-Schema ID of the event (e.g. https://github.com/fewlinesco/myapp/jsonschema/application.created.json)
 // data: is the payload of the event itself
-func CreatePublisherEvent(ctx context.Context, tx *database.Tx, subject string, eventtype string, dataschema string, data interface{}) (Event, error) {
-	rawdata, err := json.Marshal(data)
-	if err != nil {
-		return Event{}, fmt.Errorf("can't marshal event: %w", err)
-	}
+func CreatePublisherEvent(ctx context.Context, db *database.DB, subject string, eventType string, source string, dataschema string, data types.JSONText) (Event, error) {
```

Comment:
B: We should accept a transaction `*database.TX` here. Since we'll mostly dispatch after we've inserted/updated/deleted something and we want to make sure the action has been registered, we will mostly create events in a transaction together with the action.

Reply:
Oh yes indeed, my bad